### PR TITLE
feat: Implement ChatBaichuan asynchronous interface

### PR DESCRIPTION
**Pull request by maang-h**
- **Description:** Add interface to `ChatBaichuan` to support asynchronous requests
    - `_agenerate` method
    - `_astream` method

### CODE REVIEW

File: libs/community/langchain_community/chat_models/baichuan.py [hunk 1]
```diff
@@ -222,14 +248,96 @@ def _stream(
...
+                        {"finish_reason": finish_reason}
+                        if finish_reason is not None
+                        else None
+                    )
+                    chunk = ChatGenerationChunk(
+                        message=chunk, generation_info=generation_info
+                    )
+                    yield chunk
+                    if run_manager:
+                        await run_manager.on_llm_new_token(chunk.text, chunk=chunk)
```
Comment (ccurme) [inline libs/community/langchain_community/chat_models/baichuan.py:315]: This callback should be invoked prior to yielding the chunk.

https://github.com/langchain-ai/langchain/issues/16913

**Review by ccurme — COMMENTED** [review]
Looks good

**Comment by maang-h** [comment]
@ccurme @baskaryan help!!!  What should I do about this error ?

**Comment by ccurme** [comment]
@maang-h can you confirm that you've run existing integration tests? (We don't run them.)

**Comment by ccurme** [comment]
> @ccurme @baskaryan help!!! What should I do about this error ?

Should be all set, unrelated CI issue.

File: libs/community/langchain_community/chat_models/baichuan.py (hunk 1 above)
Comment (maang-h) [reply libs/community/langchain_community/chat_models/baichuan.py:314]: done

**Comment by maang-h** [comment]
> @maang-h can you confirm that you've run existing integration tests? (We don't run them.)
Yeah, integration tests passed.

**Review by ccurme — APPROVED** [review]

**Pull request merged**